### ISSUE TITLE
invalid character ‘[’ in name ‘unit-foo[bar]-key.service’

### ISSUE THREAD

**Issue by ip1981** [NixOS/nixops#614]
Probably after https://github.com/NixOS/nixops/pull/400.
Maybe related to https://github.com/NixOS/nixops/issues/609.

IMHO, per-key services are more pain than gain.

```
error: while evaluating the attribute ‘buildCommand’ of the derivation ‘nixops-machines’ at /.../nixpkgs/pkgs/build-support/trivial-builders.nix:10:14:
while evaluating anonymous function at /nix/store/760k4cq5lwap0mngr1g61ijqfbv8lr3b-nixops-1.5/share/nix/nixops/eval-machine-info.nix:330:46, called from /.../nixpkgs/lib/attrsets.nix:199:52:
while evaluating the attribute ‘buildCommand’ of the derivation ‘nixos-system-wiki-16.09pre-git’ at /.../nixpkgs/nixos/modules/system/activation/top-level.nix:102:7:
while evaluating the attribute ‘sources’ of the derivation ‘etc’ at /.../nixpkgs/nixos/modules/system/etc/etc.nix:12:5:
while evaluating anonymous function at /.../nixpkgs/nixos/modules/system/etc/etc.nix:20:20, called from undefined position:
while evaluating the attribute ‘source’ at /.../nixpkgs/lib/attrsets.nix:199:44:
while evaluating anonymous function at /.../lib/modules.nix:74:45, called from /.../nixpkgs/lib/attrsets.nix:199:52:
while evaluating the attribute ‘value’ at /.../nixpkgs/lib/modules.nix:291:9:
while evaluating the option `environment.etc.systemd/system.source':
while evaluating the attribute ‘mergedValue’ at /.../nixpkgs/lib/modules.nix:318:5:
while evaluating anonymous function at /.../nixpkgs/lib/modules.nix:318:32, called from /.../nixpkgs/lib/modules.nix:318:19:
while evaluating ‘check’ at /.../nixpkgs/lib/types.nix:110:15, called from /.../nixpkgs/lib/modules.nix:319:10:
while evaluating the attribute ‘buildCommand’ of the derivation ‘system-units’ at /.../nixpkgs/pkgs/build-support/trivial-builders.nix:10:14:
invalid character ‘[’ in name ‘unit-auto-backup-upload.s3cfg[filebackup]-key.service’
```

**Comment by rbvermaa** [comment]
@ryanartecona Any chance you could have a look at this, as you created the PR that introduced this feature?

**Comment by domenkozar** [comment]
we should just strip anything that's not allowed in unit names.

**Comment by domenkozar** [comment]
Fixed in NixOS master and 17.03
